vraptor-core/src/main/java/br/com/caelum/vraptor/observer/ExecuteMethod.java

```diff
@@ -79,6 +85,11 @@ public Void call() throws Exception {
 
 				log.debug("Invoking {}", reflectionMethod);
 				Object instance = event.getControllerInstance();
+				
+				Class<?> declaringClass = reflectionMethod.getDeclaringClass();
+				if (!declaringClass.isAssignableFrom(instance.getClass())) {
+					instance = container.instanceFor(declaringClass);
```

Comment:
Could you reproduce the bug on your machine? What is the declaringClass when the bad forward occurs?